site-src/guides/migrating-from-ingress.md

```diff
@@ -0,0 +1,261 @@
+
+### Assumptions
+
+The example includes the following assumptions:
+
+* All resources belong to the same namespace.
+* The Ingress controller:
+    * Has the corresponding IngressClass resource  `prod` in the cluster.
+    * Supports the TLS redirect feature via the `example-ingress-controller.example.org/tls-redirect` annotation. 
+* The Gateway implementation has the corresponding GatewayClass resource `prod` in the cluster.
+
+Additionally, the content of the referenced Secret and Services as well as IngressClass and GatewayClass are omitted for brevity. 
+
+### Ingress Resource
+
+The Ingress below defines the following configuration:
+
+* Configure a TLS redirect for any HTTP request for  `foo.example.com` and `bar.example.com` hostnames using the  `example-ingress-controller.example.org/tls-redirect` annotation.
+* Terminate TLS for the `foo.example.com` and `bar.example.com` hostnames using the TLS certificate and key from the Secret `example-com`.
+* Route HTTPS requests for the `foo.example.com` hostname with the URI prefix `/orders` to the `foo-orders-app` Service.
+* Route HTTPS requests for the  `foo.example.com` hostname with any other prefix to the `foo-app` Service. 
+* Route HTTPS requests for the `bar.example.com` hostname with any URI to the `bar-app` Service. 
+
+```yaml
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: example-ingress
+  annotations:
+    some-ingress-controller.example.org/tls-redirect: "True"
+spec:
+  ingressClassName: prod 
+  tls:
+  - hosts:
+    - foo.example.com
+    - bar.example.com
+    secretName: example-com
+  rules:
+  - host: foo.example.com
+    http:
+      paths:
+      - path: /
+        pathType: Prefix
+        backend:
+          service:
+            name: foo-app
+            port:
+              number: 80
+      - path: /orders
+        pathType: Prefix 
+        backend:
+          service:
+            name: foo-orders-app
+            port:
+              number: 80
+  - host: bar.example.com
+    http:
+      paths:
+      - path: /
+        pathType: Prefix 
+        backend:
+          service:
+            name: bar-app
+            port:
+              number: 80
+```
+
+The next three sections convert that Ingress into Gateway API resources.
+
+### Conversion Step 1 - Define Gateway 
+
+The following Gateway resource:
+
+* Belongs to our GatewayClass `prod`.
+* Provisions load balancing infrastructure (this depends on the Gateway implementation).
+* Configures HTTP and HTTPS listeners (entry points), which the Ingress resource included implicitly:
+    * The HTTP listener `http` on port `80`. 
+    * The HTTPS listener `https` on port `443` with TLS termination with the cert and key stored in the `example-com` Secret. It is the same Secret used in the Ingress. 
+
+Also, note that both listeners allow all HTTPRoutes from the same namespace (which is the default setting) and restrict HTTPRoute hostnames to the `example.com` subdomain.
```

Comment:
if we include this detail and the `*.example.com` configuration we should describe how this and the hostname configuration on HTTPRoutes works (intersection etc.)